File: Python3/First Unique Character in a String.py
#1
class Solution:
    def firstUniqChar(self, s: str) -> int:
        m = {}
        res = -1
        for i in range(len(s)):
            if s[i] in m:
                m[s[i]] = m[s[i]] + 1
            else:
                m[s[i]] = 1

        for i in range(len(s)-1,-1,-1):
            if m[s[i]] == 1:
                res = i

        return res
            
            
#2
class Solution:
    def firstUniqChar(self, s):
        letters='abcdefghijklmnopqrstuvwxyz'
        index=[s.index(l) for l in letters if s.count(l) == 1]
        return min(index) if len(index) > 0 else -1
            
#3
class Solution:
    def firstUniqChar(self, s):
        d = {}
        seen = set()
        for idx, c in enumerate(s):
            if c not in seen:
                d[c] = idx
                seen.add(c)
            elif c in d:
                del d[c]
        return min(d.values()) if d else -1

File: Golang/First Unique Character in a String.go
//1
func firstUniqChar(s string) int {
	d := map[byte]int{}

	for i := 0; i < len(s); i++ {
		d[s[i]]++
	}

	for i := 0; i < len(s); i++ {
		if d[s[i]] == 1 {
			return i
		}
	}

	return -1
}

//2
func firstUniqChar(s string) int {
	symbols := [26]int16{}

    for i := range s {
        symbols[s[i]-'a']++
    }
    for i := range s {
        if symbols[s[i]-'a'] == 1 {
            return i
        }
    }
    return -1
}